src/marshmallow/fields.py

```diff
@@ -166,7 +168,24 @@ def __init__(
         metadata: typing.Optional[typing.Mapping[str, typing.Any]] = None,
         **additional_metadata
     ) -> None:
-        self.default = default
+        # handle deprecated `default` and `missing` parameters
+        if dump_default is missing_ and default is not missing_:
```

Comment:
Why not just this?

    if default is not missing_:

Do we want to accept setting both to let a lib write both as a means to manage both ma versions?

Wouldn't it be better to only accept the new argument and always warn if the deprecated one is passed?